extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -0,0 +1,713 @@
+    this.container_.appendChild(this.rightMask_);
+    this.container_.appendChild(this.leftMask_);
+
+    this.rightMask_.classList.add('i-amphtml-image-slider-right-mask');
+
+    if (this.rightLabel_) {
+      this.rightLabelWrapper_ = this.doc_.createElement('div');
+      this.rightLabelWrapper_.classList
+          .add('i-amphtml-image-slider-right-label-wrapper');
+      this.rightLabel_.classList.add('i-amphtml-image-slider-right-label');
+      this.rightLabelWrapper_.appendChild(this.rightLabel_);
+      this.rightMask_.appendChild(this.rightLabelWrapper_);
+    }
+
+    this.leftMask_.classList.add('i-amphtml-image-slider-left-mask');
+    this.leftAmpImage_.classList.add('i-amphtml-image-slider-image-on-top');
+
+    if (this.leftLabel_) {
+      this.leftLabelWrapper_ = this.doc_.createElement('div');
+      this.leftLabelWrapper_.classList
+          .add('i-amphtml-image-slider-left-label-wrapper');
+      this.leftLabel_.classList.add('i-amphtml-image-slider-left-label');
+      this.leftLabelWrapper_.appendChild(this.leftLabel_);
+      this.leftMask_.appendChild(this.leftLabelWrapper_);
+    }
+  }
+
+  /**
+   * Build bar
+   * @private
+   */
+  buildBar_() {
+    this.bar_ = this.doc_.createElement('div');
+    this.barStick_ = this.doc_.createElement('div');
+    this.container_.appendChild(this.bar_);
+    this.bar_.appendChild(this.barStick_);
+
+    this.bar_.classList.add('i-amphtml-image-slider-bar');
+    this.barStick_.classList.add('i-amphtml-image-slider-bar-stick');
+  }
+
+  /**
+   * Build hint
+   * @private
+   */
+  buildHint_() {
+    if (this.hint_) {
+      this.hint_.parentNode.removeChild(this.hint_);
+    } else {
+      this.hint_ = this.doc_.createElement('div');
+    }
+    if (this.disableHint_) {
+      this.hint_ = null;
+      return;
+    }
+
+    if (this.hint_.hasAttribute('hint-loop')) {
+      this.shouldHintLoop_ = true;
+    }
+
+    if (this.hint_.hasAttribute('hint-inactive-interval')) {
+      this.hintInactiveInterval_ =
+          Number(this.hint_.getAttribute('hint-inactive-interval')) ||
+          this.hintInactiveInterval_;
+    }
+
+    const hintIcon = htmlFor(this.doc_)
+    `<div class="i-amphtml-image-slider-hint-icon">← →</div>`;
+
+    for (let i = 0; i < this.hint_.classList.length; i++) {
+      hintIcon.classList.add(this.hint_.classList[i]);
+    }
+    this.hint_.appendChild(hintIcon);
+    this.hint_.className = '';
+    this.hint_.classList.add('i-amphtml-image-slider-hint');
+    setStyles(dev().assertElement(this.hint_), {
+      opacity: 0.7,
+    });
+
+    this.barStick_.appendChild(this.hint_);
+  }
+
+  /**
+   * Install gestures for touch
+   * @private
+   */
+  registerTouchGestures_() {
+    if (this.gestures_) {
+      return;
+    }
+
+    this.gestures_ = Gestures.get(this.element);
+
+    this.gestures_.onGesture(SwipeXRecognizer, e => {
+      // We need the initial offset, yet gesture event seems not providing
```

Comment:
You can merge or rebase off of master now since https://github.com/ampproject/amphtml/pull/17125 is merged.